### PR TITLE
Sliders, Toggles and improvements

### PR DESCRIPTION
**Pull request by efosamark**

### CODE REVIEW

File: source/utils.hpp [hunk 1]
```diff
@@ -267,6 +267,7 @@ int interpretAndExecuteCommand(const std::vector<std::vector<std::string>>& comm
 
     for (auto& unmodifiedCommand : commands) {
         
+        log("command to  perform = %s", unmodifiedCommand[0].c_str());
```
Comment (i3sey) [inline source/utils.hpp:270]: uncommeted log

File: source/utils.hpp [hunk 2]
```diff
@@ -433,6 +433,7 @@ int interpretAndExecuteCommand(const std::vector<std::vector<std::string>>& comm
 
         } else if (commandName == "set-ini-val" || commandName == "set-ini-value") {
             // Edit command
+            log ("command.size() = %d", int(command.size()));
```
Comment (i3sey) [inline source/utils.hpp:436]: uncommeted log

**Review by i3sey — CHANGES_REQUESTED** [review]

File: source/ini_funcs.hpp [hunk 3]
```diff
@@ -587,7 +587,7 @@ std::string readIniValue(std::string filePath, std::string section, std::string
     return ""; // Key not found
 }
 
- std::vector<std::vector<int>> parseIntIniData (std::string input, bool skipFirstItem = true) {
+std::vector<std::vector<int>> parseIntIniData (std::string input, bool skipFirstItem = true) {
```
Comment (IrneRacoonovich) [inline source/ini_funcs.hpp:590]:
```suggestion
std::vector<std::vector<int>> parseIntIniData (const std::string& input, bool skipFirstItem = true) {
```

File: source/hex_funcs.hpp [hunk 4]
```diff
@@ -125,6 +125,55 @@ std::string readHexDataAtOffset(const std::string& filePath, const std::string&
...
+    FILE* file = fopen(filePath.c_str(), "rb");
+    if (!file) {
+        log("Failed to open the file.");
+        return nullptr;
+    }
+    return file;
+}
+
+void closeFile(FILE* const file) {
+    fclose(file);
```
Comment (IrneRacoonovich) [inline source/hex_funcs.hpp:140]: I'm not sure this function is necessary instead of just regular `fclose`.

File: source/hex_funcs.hpp [hunk 5]
```diff
@@ -125,6 +125,55 @@ std::string readHexDataAtOffset(const std::string& filePath, const std::string&
...
+}
+
+size_t findCustOffset(FILE* const file) {
+    const std::vector<std::size_t> dataOffsets = findHexDataOffsetsF(file, "43555354");
+    if (dataOffsets.empty()) {
+        log("readHexDataAtOffset: data \"%s\" not found.", "CUST");
+        return -1;
```
Comment (IrneRacoonovich) [inline source/hex_funcs.hpp:147]: `size_t` is an unsigned type. The value will not be -1 but converted to `size_t`. If you intend to return a negative value, use `long` instead.

**Pull request merged**